gcp-observability/src/main/java/io/grpc/gcp/observability/logging/GcpLogSink.java

```diff
@@ -127,6 +130,21 @@ public void write(GrpcLogRecord logProto) {
     }
   }
 
+  @VisibleForTesting
+  static Map<String, String> getCustomTags(Map<String, String> customTags,
+      Map<String, String> locationTags, String destinationProjectId) {
+    Map<String, String> tags = new HashMap<>();
+    String sourceProjectId = locationTags.get("project_id");
+    if (!Objects.equals(sourceProjectId, destinationProjectId)
+        && !Strings.isNullOrEmpty(destinationProjectId)) {
+      tags.put("source_project_id", sourceProjectId);
+    }
+    if (customTags != null) {
+      tags.putAll(customTags);
+    }
+    return tags;
```

Comment:
we are returning a "mutable" map instead of the immutable one previously.

Reply:
Updated to return `ImmutableMap<String, String>`